pkgs/applications/altcoins/zcash/default.nix

```diff
@@ -0,0 +1,49 @@
+{ stdenv, libsodium, fetchFromGitHub, wget, pkgconfig, autoreconfHook, openssl, db62, boost
+, zlib, gtest, gmock, miniupnpc, callPackage, gmp, qt4, utillinux, protobuf, qrencode, libevent
+, withGui }:
+
+let snark        = callPackage ./snark { inherit boost openssl; };
```

Comment:
Sorry if I'm not understanding you clearly, what's the reason for not adding `snark` to `nixpkgs` top-level? If it would be useful outside of `zcash` then it would be nice to be able to use it without needing `zcash`. I might not be understanding the dependency between these two so ignore me if that's the case.